server/routes/companies.js

```diff
@@ -2,14 +2,174 @@
+  }
+  else if(pid == undefined && mid != undefined){
+    db.query('SELECT company_id FROM company_major WHERE major_id = $1;', [mid], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+  } else if(pid != undefined && mid == undefined){
+    db.query('SELECT company_id FROM company_position WHERE position_id = $1;', [pid], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+  } else{ //both
+      db.query('SELECT DISTINCT company_major.company_id FROM company_major LEFT OUTER JOIN company_position ON (position_id = $1 AND major_id = $2);', [pid, mid], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+    }
+});
+
+//Get all companies containing term substring in their name
+router.get('/search', function(req,res,next) {
+  const search = req.query.term;
+  if(search == undefined){
+    res.send('Not Found');
+  }
+
+  db.query('SELECT id FROM company WHERE name ILIKE $1', ['%' + search +'%'], (err, result) => {
+    if(err){
+      return next(err);
+    }
+    res.send(result.rows);
+  });
+
+});
+
+//Get all companies' info
 router.get('/', function(req, res, next) {
+  db.query('SELECT id FROM company;', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all company names
+router.get('/names', function(req, res, next) {
+  db.query('SELECT id, name FROM company ORDER BY name;', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all company names and logos
+router.get('/logos', function(req, res, next) {
+  db.query('SELECT id, name, logo FROM company ORDER BY name;', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+// Get all company names, logos, and websites
+router.get('/websites', function(req, res, next) {
+  db.query('SELECT id, name, logo, website FROM company ORDER BY name;', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get company info by company_id
+router.get('/:company_id', function(req, res, next) {
+  const id = req.params.company_id;
+  db.query('SELECT * FROM company WHERE id = $1;', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all events associated with a given company
+router.get('/:company_id/events', function(req, res, next) {
+  const id = req.params.company_id;
+  db.query('SELECT * FROM event_company WHERE company_id= $1;', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get all positions a given company is hiring
+router.get('/:company_id/positions', function(req, res, next) {
+    // TODO
+    const id = req.params.company_id;
```

Comment:
Make sure that general comments should be applied throughout code, not just at the one instance noted by a comment (e.g. rename id variable to company_id here, use SELECT position_id, etc.)